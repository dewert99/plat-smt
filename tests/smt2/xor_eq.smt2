(declare-const a Bool)
(declare-const b Bool)
(declare-const c Bool)
(push)
(assert (= a (xor b c)))
(check-sat-assuming (b c))
(get-model)
(pop)
(push)
(assert (= a (xor b true)))
(assert (= c (not (xor false a))))
(check-sat-assuming (b))
(get-model)
(pop)
(assert (xor a false))
(assert (not (xor true b)))
(assert (= c (xor true false)))
(check-sat)
(get-model)
(assert (xor true false))
(check-sat)
(assert (xor false false))
(check-sat)