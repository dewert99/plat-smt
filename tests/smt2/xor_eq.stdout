sat
(
 (define-fun a () Bool false)
 (define-fun b () Bool true)
 (define-fun c () Bool true)
)
sat
(
 (define-fun a () Bool false)
 (define-fun b () Bool true)
 (define-fun c () Bool true)
)
sat
(
 (define-fun a () Bool true)
 (define-fun b () Bool true)
 (define-fun c () Bool true)
)
sat
unsat
